tests/ModelContextProtocol.Tests/Client/McpClientExtensionsTests.cs

```diff
@@ -340,15 +340,6 @@ public async Task SendNotificationAsync_HonorsJsonSerializerOptions()
         await Assert.ThrowsAsync<NotSupportedException>(() => client.SendNotificationAsync("Method4", new { Value = 42 }, emptyOptions, cancellationToken: TestContext.Current.CancellationToken));
     }
 
-    [Fact]
-    public async Task GetPromptsAsync_HonorsJsonSerializerOptions()
-    {
-        JsonSerializerOptions emptyOptions = new() { TypeInfoResolver = JsonTypeInfoResolver.Combine() };
-        IMcpClient client = await CreateMcpClientForServer();
-
-        await Assert.ThrowsAsync<NotSupportedException>(() => client.GetPromptAsync("Prompt", new Dictionary<string, object?> { ["i"] = 42 }, emptyOptions, cancellationToken: TestContext.Current.CancellationToken));
```

Comment:
What was this trying to test before? Does the NotSupportedException somehow indicate that the passed-in JsonSerializerOptions? And are we deleting this because we have coverage elsewhere?

Reply:
Previously the implementation was using `serializerOptions.GetTypeInfo<IReadOnlyDictionary<string, object?>>()` to get a serializer for the arguments. Now because it's using the GetPromptRequestParams, it's using the McpJsonUtilities source generated code.

@eiriktsarpalis, is this the right or wrong thing to do?

Reply:
The test is meant to check that user-supplied JSO's are being respected when using it with user-supplied objects. It should be kept but I think it also needs to incroporate this change:

https://github.com/modelcontextprotocol/csharp-sdk/pull/224/files#r2030879138

Reply:
> It should be kept

Why? Especially if the arguments is changed to use a jsonelement instead of object, what user object is there here+

Reply:
The `GetPromptAsync` signature is unchanged in that it accepts an `IReadOnlyDictionary<string, object?>?` parameter which contains user-provided values. As such, we should preserve test coverage that links the user-provided JSO to the serialization of user-provided objects.